src/openforms/submissions/tests/form_logic/test_modify_components.py

```diff
@@ -1046,6 +1046,57 @@ def test_checkbox_frontend_logic(self):
 
         self.assertEqual(submission_step.data["textField"], "Test value")
 
+    @tag("gh-3964")
+    def test_number_frontend_logic(self):
+        form = FormFactory.create()
+        step = FormStepFactory.create(
+            form=form,
+            form_definition__configuration={
+                "components": [
+                    {
+                        "key": "number",
+                        "type": "number",
+                    },
+                    {
+                        "key": "currency",
+                        "type": "currency",
+                    },
+                    {
+                        "key": "textField1",
+                        "type": "textfield",
+                        "clearOnHide": True,
+                        "conditional": {"eq": "1", "show": True, "when": "number"},
+                    },
+                    {
+                        "key": "textField2",
+                        "type": "textfield",
+                        "clearOnHide": True,
+                        "conditional": {"eq": "1", "show": True, "when": "currency"},
```

Comment:
```suggestion
                        "conditional": {"eq": "1.0", "show": True, "when": "currency"},
```
Maybe a test with a `float`?

Reply:
I was worried of potential flakiness due to comparing floats :thinking:

Reply:
I think it should be alright, as long as you don't perform any calculations on the floating numbers

And if we have flakyness in the test we will have it at runtime as well so I think this is a good thing :)

Reply:
Yes! Although I remeber at some point we had discouraged clients to do comparisons on with the floating points, so hopefully it shouldn't happen too often.

Reply:
If this becomes an issue we could probably use decimals to do the comparison